packages/ui-react/src/containers/ShelfList/ShelfList.tsx

```diff
@@ -61,28 +65,31 @@ const ShelfList = ({ rows }: Props) => {
         loader={<InfiniteScrollLoader key="loader" />}
         useWindow={false}
       >
-        {rows.slice(0, rowsToLoad).map(({ type, featured, title }, index) => {
+        {rows.slice(0, rowsToLoad).map(({ type, featured, title, custom }, index) => {
           const { data: playlist, isPlaceholderData, error } = playlists[index];
 
           if (!playlist?.playlist?.length) return null;
 
           const posterAspect = parseAspectRatio(playlist.cardImageAspectRatio || playlist.shelfImageAspectRatio);
           const visibleTilesDelta = parseTilesDelta(posterAspect);
 
+          const translatedKey = custom?.[`title-${language}`];
+          const translatedTitle = translatedKey ? (translatedKey as string) : title || playlist?.title;
+
```

Comment:
Instead of casting here, we should type check or update the validation schema to catch this so we can use `Record<string, string>` instead.

```suggestion
          const translatedTitle = typeof translatedKey === 'string' ? translatedKey : title || playlist?.title;

```